src/Features/Core/Portable/MakeMethodStatic/AbstractMakeMethodStaticDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.MakeMethodStatic
+{
+    internal abstract class AbstractMakeMethodStaticDiagnosticAnalyzer<TSyntaxKind, TMethodDeclarationSyntax>
+        : AbstractCodeStyleDiagnosticAnalyzer
+        where TSyntaxKind : struct
+        where TMethodDeclarationSyntax : SyntaxNode
+    {
+        protected AbstractMakeMethodStaticDiagnosticAnalyzer(LocalizableString title, LocalizableString message)
+            : base(IDEDiagnosticIds.MakeMethodStaticDiagnosticId, title, message)
+        {
+        }
+
+        protected sealed override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxNodeAction(AnalyzeNode, GetMethodDeclarationSyntaxKind());
+
+        public sealed override bool OpenFileOnly(Workspace workspace) => true;
+        public sealed override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var declaration = (TMethodDeclarationSyntax)context.Node;
+            var semanticModel = context.SemanticModel;
+            var cancellationToken = context.CancellationToken;
+
+            var methodSymbol = semanticModel.GetDeclaredSymbol(declaration, cancellationToken);
+            if (methodSymbol == null)
+            {
+                return;
+            }
+
+            if (methodSymbol.IsStatic ||
+                methodSymbol.DeclaredAccessibility != Accessibility.Private)
```

Comment:
Might want to have options to control the 'private' bit.  For example, someone might want to opt into knowing about public methods they have that can be private.